src/pages/home-page/model/store.ts

```diff
@@ -0,0 +1,12 @@
+import { createEvent, createStore } from 'effector';
+
+export const setUsername = createEvent<string>();
+export const setPassword = createEvent<string>();
+
+export const $usernameStore = createStore('').on(setUsername, (_, value) => value);
```

Comment:
consider separating the store from its' reaction .on

`export const $usernameStore = createStore('')`

`$usernameStore.on(setUserName, (_, username) => username)`